.github/scripts/branch_pr_issue_closer.py

```diff
@@ -135,176 +98,162 @@ def _main_prog():
-    #which ends with either a space, a comma, a period, or
-    #the end of the string itself:
-    issue_pattern = re.compile(r'#[0-9]+(\s|,|$)|.')
+        #create issue pattern ("the number symbol {#} + a number"),
+        #which ends with either a space, a comma, a period, or
+        #the end of the string itself:
+        issue_pattern = re.compile(r'#[0-9]+(\s|,|$)|.')
 
-    #Create new "close" issues list:
-    close_issues = []
+        #Create new "close" issues list:
+        close_issues = []
 
-    #Create new "closed" PR list:
-    close_pulls = []
+        #Create new "closed" PR list:
+        close_pulls = []
 
-    #Search text right after keywords for possible issue numbers:
-    for match in word_matches:
+        #Search text right after keywords for possible issue numbers:
+        for match in word_matches:
 
-        #create temporary string starting at end of match:
-        tmp_msg_str = pr_msg_lower[match.end():]
+            #create temporary string starting at end of match:
+            tmp_msg_str = pr_msg_lower[match.end():]
 
-        #Check if first word matches issue pattern:
-        if issue_pattern.match(tmp_msg_str) is not None:
+            #Check if first word matches issue pattern:
+            if issue_pattern.match(tmp_msg_str) is not None:
 
-            #If so, then look for an issue number immediately following
-            first_word = tmp_msg_str.split()[0]
+                #If so, then look for an issue number immediately following
+                first_word = tmp_msg_str.split()[0]
 
-            #Extract issue number from first word:
-            try:
-                #First try assuming the string is just a number
-                issue_num = int(first_word[1:]) #ignore "#" symbol
-            except ValueError:
-                #If not, then ignore last letter:
+                #Extract issue number from first word:
                 try:
-                    issue_num = int(first_word[1:-1])
+                    #First try assuming the string is just a number
+                    issue_num = int(first_word[1:]) #ignore "#" symbol
                 except ValueError:
-                    #If ignoring the first and last letter doesn't work,
-                    #then the match was likely a false positive,
-                    #so set the issue number to one that will never be found:
-                    issue_num = -9999
-
-            #Check if number is actually for a PR (as opposed to an issue):
-            if issue_num in open_pulls:
-                #Add PR number to "close pulls" list:
-                close_pulls.append(issue_num)
-            elif issue_num in open_issues:
-                #If in fact an issue, then add to "close issues" list:
-                close_issues.append(issue_num)
+                    #If not, then ignore last letter:
+                    try:
+                        issue_num = int(first_word[1:-1])
```

Comment:
I know this isn't new with this PR, but what is the use case for this? (why ignore the last letter?)

Reply:
This is because the regex used to extract the `first_word` variable allows for it to end with a space, period, or comma, and so for situations where there is a comma or period you'll want to ignore the last character.  I believe this was done so that users could write lists like the following in their PR description:

`Fixes #X, Closes #Y, Resolves #Z.`

Of course if you think instead we should require only a space (or nothing) at the end then I can change the regex and remove this `try` statement.  Just let me know!

Reply:
ahhh punctuation! of course!

that makes sense - thanks. Can you perhaps add a comment explaining that for the next person (possibly me again) who is confused?

Reply:
Ahh yes, I have heard that those "comment" things you mentioned are useful.  Comment added!

Reply:
thank you!!